vortex-array/src/array/primitive/compute/take.rs

```diff
@@ -43,19 +43,24 @@ impl TakeFn<PrimitiveArray> for PrimitiveEncoding {
 // In which case, Rust should reuse the same Vec<u64> the result.
 fn take_primitive<T: NativePType, I: NativePType + AsPrimitive<usize>>(
     array: &[T],
-    indices: Vec<I>,
+    indices: &[I],
```

Comment:
Maybe the optimisation isn't worth the copies for the non-u65 cases, but you should update the comment.

Reply:
AFAICT, we never avoided a copy because we cloned the indices incrementing the ref count by one. I've added a new comment which describes what we could do if the method signature changes.

Reply:
Ah, but in some cases we then decompress when we call into_primitive (which calls into_canonical).